### ISSUE TITLE
double underscore in filename turns into markdown bold format

### ISSUE THREAD

**Issue by ukeSJTU** [cline/cline#1028]
### What happened?

When working with cline on a python project, cline treats the double underscore as bold text in markdown, however it is actually part of the filename such as `__init__.py`.

See the screenshot below for what I mean:

![截屏2024-12-25 12 52 15](https://github.com/user-attachments/assets/4cee3df3-28e1-44ca-81df-339ceb5fbecd)

### Steps to reproduce

1. Select model type: anthropic/claude-3.5-sonnet
2. Ask cline to: Create a new `__filename__.py` file and print("Hello python")

**Comment by saoudrizwan** [comment]
Good catch, I'm using remark/rehype under the hood so we could probably create a plugin that bypasses processing on filenames using regex.

**Comment by jpdjere** [comment]
@saoudrizwan Any reason this issue wasn't closed by https://github.com/cline/cline/pull/1070 ?

**Comment by mgssf** [comment]
It looks like this issue is resolved in v3.11.1:

![Image](https://github.com/user-attachments/assets/3e871b16-27cc-44de-9b7b-f31e4f6de1a8)

**Comment by char8x** [comment]
@ukeSJTU Are you still troubled by this issue? If it's resolved, you can close it.
